MODULE.bazel

```
@@ -0,0 +1,11 @@
+bazel_dep(name = "gazelle", version = "0.40.0")
+bazel_dep(name = "rules_go", version = "0.50.1")
+
+go_sdk = use_extension("@rules_go//go:extensions.bzl", "go_sdk")
+go_sdk.download(version = "1.22.7")
+
+go_deps = use_extension("@gazelle//:extensions.bzl", "go_deps")
+go_deps.from_file(go_mod = "//:go.mod")
+
```

I've cracked it! It's ignoring all the overrides and fixes being attempted in `deps.bzl` because _this_ is using BzlMod instead!

Here's how I think we can fix it:

1. Get rid of `deps.bzl`, `WORKSPACE.bazel`, and the `gazelle` rule in `BUILD.bazel`
2. Add a gazelle [override](https://github.com/bazel-contrib/rules_go/blob/master/docs/go/core/bzlmod.md#overrides) here:
```suggestion
go_deps = use_extension("@gazelle//:extensions.bzl", "go_deps")
go_deps.from_file(go_mod = "//:go.mod")

go_deps.gazelle_override(
    directives = [
        "gazelle:go_generate_proto false",
    ],
    path = "github.com/DataDog/sketches-go",
)
```

With that I can successfully `bazel build //...` !